spot_rl_experiments/spot_rl/real_policy.py

```diff
@@ -32,68 +47,133 @@ def __init__(
-            action_space=action_space,
-        )
-        print("Actor-critic architecture:", self.policy)
-        # Move it to the device
-        self.policy.to(self.device)
+        # Load the config
+        if not is_hab3_policy:
+            config = checkpoint["config"]
 
-        # Load trained weights into the policy
-        self.policy.load_state_dict(
-            {k[len("actor_critic.") :]: v for k, v in checkpoint["state_dict"].items()}
-        )
+        """ Disable observation transforms for real world experiments """
+        if not is_hab3_policy:
+            config.defrost()
+            config.RL.POLICY.OBS_TRANSFORMS.ENABLED_TRANSFORMS = []
+            config.freeze()
+            config.RL.POLICY["init"] = False
+
+            self.policy = policy_class.from_config(
+                config=config,
+                observation_space=observation_space,
+                action_space=action_space,
+            )
+            print("Actor-critic architecture:", self.policy)
+            # Move it to the device
+            self.policy.to(self.device)
+
+            # Load trained weights into the policy
+            self.policy.load_state_dict(
+                {
+                    k[len("actor_critic.") :]: v
+                    for k, v in checkpoint["state_dict"].items()
+                }
+            )
 
         self.prev_actions = None
         self.test_recurrent_hidden_states = None
         self.not_done_masks = None
-        self.config = config
+        if is_hab3_policy:
+            self.config = None
+        else:
+            self.config = config
         self.num_actions = action_space.shape[0]
         self.reset_ran = False
+        self.is_hab3_policy = is_hab3_policy
         print("Policy loaded.")
 
     def reset(self):
         self.reset_ran = True
-        self.test_recurrent_hidden_states = torch.zeros(
-            1,  # The number of environments. Just one for real world.
-            self.policy.net.num_recurrent_layers,
-            self.config.RL.PPO.hidden_size,
-            device=self.device,
-        )
+        if not self.is_hab3_policy:
+            self.test_recurrent_hidden_states = torch.zeros(
+                1,  # The number of environments. Just one for real world.
+                self.policy.net.num_recurrent_layers,
+                self.config.RL.PPO.hidden_size,
+                device=self.device,
+            )
+        else:
+            self.test_recurrent_hidden_states = torch.zeros(
+                1,  # The number of environments. Just one for real world.
+                4,
+                512,
+                device=self.device,
+            )
 
         # We start an episode with 'done' being True (0 for 'not_done')
         self.not_done_masks = torch.zeros(1, 1, dtype=torch.bool, device=self.device)
         self.prev_actions = torch.zeros(1, self.num_actions, device=self.device)
 
+    def get_action(self, mu_maybe_std, std):
```

Comment:
Confusing arg-name? `mu_maybe_std`?

Reply:
@jimmytyyang  might be able to comment on this, if its okay then I shall rename mu_maybe_std with mu

Reply:
Hi all, this is to follow the convention from hab3